### DISCUSSION TITLE
Can Route Handlers return RSCs?

### DISCUSSION THREAD

**Discussion by steve-marmalade** [vercel/next.js#60523] [App Router]
I am looking for a way to fetch RSCs from the client in order to implement infinite scroll. In particular, I want all of the benefits of RSCs like the React Cache and the Fetch Cache, as well as streaming support via Suspense boundaries.

Server Actions almost work for this case, but they no longer seem to be recommended for data fetching (only for mutation) and they also don't leverage the React Cache, which is critical for deduping requests.

**Comment by steve-marmalade** [comment]
This discussion seems very relevant https://github.com/vercel/next.js/discussions/58674, and from what I gather there is no way to do this currently

**Comment by sim391calado** [comment]
> I am looking for a way to fetch RSCs from the client in order to implement infinite scroll. In particular, I want all of the benefits of RSCs like the React Cache and the Fetch Cache, as well as streaming support via Suspense boundaries.
> 
> Server Actions almost work for this case, but they no longer seem to be recommended for data fetching (only for mutation) and they also don't leverage the React Cache, which is critical for deduping requests.

I'm also looking for a solution for this. Did you find anything relevant?
